app/controllers/api/v1/release_engines/npm/package_metadata_controller.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+require 'compact_index'
+
+module Api::V1::ReleaseEngines
+  class Npm::PackageMetadataController < Api::V1::BaseController
+    before_action :scope_to_current_account!
+    before_action :require_active_subscription!
+    before_action :authenticate_with_token
+    before_action :set_package
+
+    def show
+      authorize! package,
+        to: :show?
+
+      artifacts = authorized_scope(package.artifacts.npm_package_tgz.order_by_version)
+        .where_assoc_exists(:manifest) # must exist
+        .preload(:manifest,
+          release: %i[product entitlements constraints],
+        )
+      authorize! artifacts,
+        to: :index?
+
+      latest   = artifacts.first
+      metadata = artifacts.reduce(name: package.name, time: {}, 'dist-tags': { latest: latest.release.version }, versions: {}) do |metadata, artifact|
+        package_json = artifact.manifest.as_package_json
+        release      = artifact.release
+
+        metadata[:time][release.version]     = artifact.created_at.iso8601
+        metadata[:'dist-tags'][release.tag]  = release.version if release.tag?
+        metadata[:versions][release.version] = package_json.merge(
+          dist: {
+            tarball: vanity_v1_account_release_artifact_url(current_account, artifact, filename: artifact.filename, host: request.host),
```

Comment:
Would be nice to have a shasum here but afaik npm only supports SHA1, so we'd need to update the CLI to support it.